dashboard/app/controllers/api_controller.rb

```diff
@@ -100,7 +99,7 @@ def section_progress
         {
           class: level_class,
           title: title,
-          url: build_script_level_url(script_level, section_id: @section.id, user_id: student.id)
+          url: build_script_level_url(script_level, section_id: section.id, user_id: student.id)
         }
```

Comment:
As an aside, it's unfortunate there's so much logic here in the controller. Ideally this would all be exposed in the model(s) and tested there.